### PR TITLE
Implement Windows service directly, handle systemd stop signal

### PR DESCRIPTION
**Pull request by adjackura**

### CODE REVIEW

File: google-osconfig-agent.service [hunk 1]
```diff
@@ -9,6 +9,8 @@ Restart=always
 RestartSec=1
 StartLimitInterval=120
 StartLimitBurst=3
+KillMode=mixed
+KillSignal=SIGTERM
```
Comment (hopkiw) [inline google-osconfig-agent.service]: this is the default for `KillMode=mixed`, right?

File: google-osconfig-agent.service [hunk 2]
```diff
@@ -9,6 +9,8 @@ Restart=always
 RestartSec=1
 StartLimitInterval=120
 StartLimitBurst=3
+KillMode=mixed
```
Comment (hopkiw) [inline google-osconfig-agent.service]: Why change this?

File: main_windows.go [hunk 3]
```diff
@@ -86,3 +90,43 @@ func obtainLock() {
...
+				status <- c.CurrentStatus
+			case svc.Stop, svc.Shutdown:
+				cncl()
+			default:
+			}
+		}
+	}
+}
+
+func runService(ctx context.Context, run func(context.Context)) {
```
Comment (hopkiw) [inline main_windows.go]: same here. we own this and are not creating a general-purpose library. don't take functions as arguments.

File: google-osconfig-agent.service (hunk 2 above)
Comment (adjackura) [reply google-osconfig-agent.service]: SIGTERM was getting sent to all processes, which was causing issues with things like dpkg and left them in a messy state. If i read the docs correctly this should only send SIGTERM to osconfig, then SIGKILL to all after the timeout

File: google-osconfig-agent.service (hunk 1 above)
Comment (adjackura) [reply google-osconfig-agent.service]: Yes, I put it in here to make it clear what signal is getting sent without having to dig through docs first.

**Pull request merged**